**Import Packages**

In [164]:
#!pip install selenium
import selenium
from selenium.webdriver.common.keys import Keys
import time
import re
from selenium import webdriver
print('Selenium Version:',selenium.__version__)

Selenium Version: 3.141.0


Using: 

    Selenium Version: 3.141.0
    Firefox Version: 60.0.1 (64-bit)
    geckodriver-v0.23.0-win64.zip

**1. Go to website**
    - For indeed I went directly to sign-in page
    - 'Inspect Element' and find the sign-in element

In [165]:
driver = webdriver.Firefox()
driver.get("https://secure.indeed.com/account/login?service=my&hl=en_US&co=US&continue=https%3A%2F%2Fwww.indeed.com%2F")
# Wait for 10 seconds
time.sleep(10)

Find Username and Password Fields

In [166]:
username = driver.find_element_by_id("signin_email")
password = driver.find_element_by_id("signin_password")

Insert Username and Passord

In [167]:
username.send_keys("myEmailAddress@email.com")#Insert Username and Password into fields
password.send_keys("myvoiceismypassportverify.")

Find Submit/Sign-in button and click

In [168]:
driver.find_element_by_class_name("icl-Button.icl-Button--primary.icl-Button--md.icl-Button--block").click()
# Wait for 5 seconds
time.sleep(5)

Find Job Searchbar

In [169]:
searchbar=driver.find_element_by_id("text-input-what")

Insert text into search field

In [170]:
searchbar.send_keys("Data Science")

Click on search link

In [171]:
driver.find_element_by_class_name("icl-Button.icl-Button--primary.icl-Button--md.icl-WhatWhere-button").click()
# Wait for 10 seconds
time.sleep(10)

Results information

In [172]:
pages=driver.find_element_by_id('searchCount')
pages.text

'Page 1 of 140 jobs'

In [173]:
linkList=driver.find_elements_by_class_name("jobsearch-SerpJobCard.row.result.clickcard")
print('Number of results on first page:',len(linkList))
# Wait for 5 seconds
time.sleep(5)

Number of results on first page: 18


Find search results

Select first search result

In [174]:
linkList=driver.find_elements_by_id('sja1')
print(linkList[0].text)

Data Engineer, AWS | Fortune 100 Retailer


In [175]:
linkList[0].click()
time.sleep(5)

In [176]:
Description=driver.find_element_by_id("vjs-desc")
print(Description.text)

BICP, a dedicated BI, Analytics & Big Data consulting firm, is currently looking to hire a NextGen Data Engineer for an engagement at our longstanding and direct Fortune 100 retail client in Portland, OR. We currently have 15+ employees on-site supporting multiple Geo’s with their drive to deploy NextGen platforms & analytics capabilities to support real-time decision making. We’re looking for candidates with 2+ years of experience with data engineering with emphasis on data analytics and reporting. Must possess expert level SQL skills and in addition to strong relational database engineering (Oracle, SQL Server, or Teradata) experience. Hands-on, enterprise level AWS, Spark, Shell and Python experience is required. Snowflake, Airflow, Autosys, GitHub and Jenkins is preferred but all are not required. Ideal consultant will be familiar with agile delivery methodologies including Scrum, SAFe or Extreme Programming. Any knowledge of building solutions with data visualization and reporting

Print/Store first search result

In [177]:
SearchResultsDict={}
SearchResultsDict['text1']=Description.text
print(SearchResultsDict['text1'])

BICP, a dedicated BI, Analytics & Big Data consulting firm, is currently looking to hire a NextGen Data Engineer for an engagement at our longstanding and direct Fortune 100 retail client in Portland, OR. We currently have 15+ employees on-site supporting multiple Geo’s with their drive to deploy NextGen platforms & analytics capabilities to support real-time decision making. We’re looking for candidates with 2+ years of experience with data engineering with emphasis on data analytics and reporting. Must possess expert level SQL skills and in addition to strong relational database engineering (Oracle, SQL Server, or Teradata) experience. Hands-on, enterprise level AWS, Spark, Shell and Python experience is required. Snowflake, Airflow, Autosys, GitHub and Jenkins is preferred but all are not required. Ideal consultant will be familiar with agile delivery methodologies including Scrum, SAFe or Extreme Programming. Any knowledge of building solutions with data visualization and reporting

Move to the next-

    Second search result

In [178]:
linkList=driver.find_elements_by_id('sja2')
print(linkList[0].text)

Reconciliation Analyst


In [179]:
linkList=driver.find_elements_by_class_name("jobsearch-SerpJobCard.row.result.clickcard")
len(linkList)

linkList[1].click()
Description=driver.find_element_by_id("vjs-desc")
print(Description.text)

Be part of a young creative company. Work with cutting-edge technology. Data Science is in the DNA of Aspen Capital and we hope it is in yours as well. Join us as we rewrite the rules of residential and commercial mortgages and real estate.
Aspen Capital is a private equity firm based in Portland, OR and New York, NY. We utilize data and technology to enhance business insight, propel growth, transform our investment strategies and business operations, and execute industry-leading deals. The unique Aspen Capital worldview is reflected in a nimble, efficient organizational structure that allows the company to capitalize on market demands, seize business opportunities and excel in a wide range of roles including investment, lending and servicing, acquisitions, management, joint ventures, asset management, recapitalization and advisory services.
Aspen Capital is currently seeking a Reconciliation Analyst. This position reviews and reconciles all interim cash collections activity associated

Move to the next set of links

In [180]:
driver.find_element_by_class_name("pn").click()
time.sleep(10)

Close dialog box

In [181]:
driver.find_element_by_class_name('icl-Icon.icl-Icon--sm.icl-Icon--black.close').click()

Pages / jobs

In [182]:
pages=driver.find_element_by_id('searchCount')
pages.text

'Page 2 of 140 jobs'

In [183]:
digits=list(re.findall('\d+', pages.text))
digits

['2', '140']

*Do a while loop where if a counter is =<140 than continue to click on links. And with each click increase the counter by 1. Meanwhile keep clicking next until 140 is reached.*

Stop Selenium

In [184]:
driver.close()
driver.quit()